client/src/main/java/org/asynchttpclient/DefaultAsyncHttpClientConfig.java

```diff
@@ -1252,7 +1254,21 @@ private ProxyServerSelector resolveProxyServerSelector() {
       return ProxyServerSelector.NO_PROXY_SELECTOR;
     }
 
+    private static Timer newNettyTimer(Builder builder) {
+      ThreadFactory threadFactory = builder.threadFactory != null ? builder.threadFactory : new DefaultThreadFactory(builder.threadPoolName + "-timer");
+
+      HashedWheelTimer timer = new HashedWheelTimer(threadFactory);
+      timer.start();
+      return timer;
+    }
+
     public DefaultAsyncHttpClientConfig build() {
+      if (nettyTimer == null)
+        nettyTimer = newNettyTimer(this);
+
+      if (cookieStore == null)
+        cookieStore = new ThreadSafeCookieStore(nettyTimer);
```

Comment:
This causes a slight behavior change: there's no longer any way to disable CookieStore, except providing a noop implementation. And it causes all the existing `if (cookieStore != null)` guards useless.